src/machine/usb/descriptor/hidreport.go

```diff
@@ -129,48 +129,66 @@ func HIDReportID(id int) []byte {
 }
 
 func HIDLogicalMinimum(min int) []byte {
-	if min > 255 {
-		return []byte{hidLogicalMinimum + 1, uint8(min), uint8(min >> 8)}
+	m := uint32(min)
+	if 65535 < m {
```

Comment:
I would suggest changing this to:

```go
switch {
        case m > 65535:
                return []byte{hidLogicalMinimum + 2, uint8(m), uint8(m >> 8), uint8(m >> 16), uint8(m >> 32)}
        case m > 255:
                return []byte{hidLogicalMinimum + 1, uint8(m), uint8(m >> 8)}
        default:
                return []byte{hidLogicalMinimum, byte(min)}
}
````

@sago35 what do you think?

Reply:
Changed to use switch-case.